### PR TITLE
Integration of App-Server Utils module to App Server 6.0.0.

### PR DESCRIPTION
**Pull request by chirangaalwis**
Integrating the refactored app-server utils module. Fixed issues related to server and web-app descriptors.

### CODE REVIEW

File: modules/appserver-utils/src/main/java/org/wso2/appserver/utils/PathUtils.java [hunk 1]
```diff
@@ -32,90 +27,50 @@
  * @since 6.0.0
  */
 public final class PathUtils {
-
     private static final Path PATH_CATALINA_BASE;
-    private static final Path PATH_CATALINA_CONFIG;
-    private static final Path PATH_WSO2_CONFIG;
+    private static final Path PATH_CATALINA_CONFIG_BASE;
+    private static final Path PATH_WSO2_CONFIG_BASE;
```
Comment (manoj-kumara) [inline modules/appserver-utils/src/main/java/org/wso2/appserver/utils/PathUtils.java]: This should be PATH_APP_SERVER_CONFIG_BASE

File: distribution/contents/conf/wso2/wso2as-web.xml [hunk 2]
```diff
@@ -11,18 +11,19 @@
...
-    </classloading>
+    </class-loader>
     <saml2-single-sign-on>
         <!--<skip-uris>
             <skip-uri>http://www.example.com</skip-uri>
         </skip-uris>-->
         <handle-consumer-url-after-slo>true</handle-consumer-url-after-slo>
         <query-params>tenant=admin&amp;dialect=SAML</query-params>
-        <application-server-url>http://localhost:8443</application-server-url>
+        <application-server-url>https://localhost:8443</application-server-url>
```
Comment (manoj-kumara) [inline distribution/contents/conf/wso2/wso2as-web.xml]: port should be 9443 right ?

File: modules/appserver-utils/src/main/java/org/wso2/appserver/configuration/listeners/ContextConfigurationLoader.java [hunk 3]
```diff
@@ -90,36 +83,35 @@ public void lifecycleEvent(LifecycleEvent lifecycleEvent) {
...
-                    .getUnmarshalledObject(defaultWebAppDescriptor, schemaPath, ContextConfiguration.class);
-
-            if (Files.exists(contextWebAppDescriptor)) {
-                ContextConfiguration local = XMLUtils
-                        .getUnmarshalledObject(contextWebAppDescriptor, schemaPath, ContextConfiguration.class);
-                effective.merge(local);
+            AppServerWebAppConfiguration effective;
+            try {
+                InputStream inputStream = context.getServletContext().getResourceAsStream(
+                        "/" + Constants.WEBAPP_RESOURCE_FOLDER + "/" + Constants.WEBAPP_DESCRIPTOR);
```
Comment (manoj-kumara) [inline modules/appserver-utils/src/main/java/org/wso2/appserver/configuration/listeners/ContextConfigurationLoader.java]: Should use file separator.

**Comment by Niranjan-K** [comment]
@chirangaalwis : Do we have Jira for this?

**Comment by chirangaalwis** [comment]
Yes, we have for the utils module. https://wso2.org/jira/browse/WSAS-2190

**Pull request merged**